library/core/src/cell.rs

```diff
@@ -2323,6 +2324,7 @@ impl<T: ?Sized> SyncUnsafeCell<T> {
     /// Gets a mutable pointer to the wrapped value.
     ///
     /// See [`UnsafeCell::get`] for details.
+    #[cfg_attr(not(bootstrap), rustc_as_ptr)]
```

Comment:
We don't actually need this one, as it
1. Just like with `MaybeUninit::slice_as_ptr`, it is not callable in `temporary.method()` form.
2. It already takes a pointer as an argument, while `dangling_pointers_from_temporaries` is all about converting from a temporary to a pointer, not pointer-to-pointer casts.

Reply:
To clarify, the lint is about crossing the boundary between borrow-checked lifetime-having Rust types and the unmanaged unchecked untracked land of raw pointers. And in particular, the specific case where "the temporary will be deallocated at the end of the statement because nothing is referencing it as far as the type system is concerned".

This function does not cross the said boundary.

Reply:
Sorry, I see the issue. Just pushed a fix.

Just to confirm my understanding, the reason this does not apply is because the `raw_get` function simply casts a constant pointer to a mutable one, and thus does not concern the crossing of the managed-unmanaged boundary, as opposed to the `get` function that returns a raw pointer to the inner value?

Reply:
Yes